swatch-contracts/src/main/java/com/redhat/swatch/contract/config/DebugClientLogger.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package com.redhat.swatch.contract.config;
+
+import io.quarkus.arc.Unremovable;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpClientRequest;
+import io.vertx.core.http.HttpClientResponse;
+import jakarta.enterprise.context.ApplicationScoped;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.jboss.resteasy.reactive.client.api.ClientLogger;
+
+/** ClientLogger implementation that only logs requests for a matching URI */
+@Slf4j
+@Unremovable
+@ApplicationScoped
+public class DebugClientLogger implements ClientLogger {
+
+  @ConfigProperty(name = "rest-client-debug-logging.uri-filter")
+  Pattern uriFilterPattern;
+
+  private ThreadLocal<Boolean> shouldLog = new ThreadLocal<Boolean>();
+
+  @Override
+  public void setBodySize(int bodySize) {
+    // intentionally ignored
+  }
+
+  @Override
+  public void logResponse(HttpClientResponse response, boolean redirect) {
+    if (Objects.equals(shouldLog.get(), Boolean.TRUE)) {
```

Comment:
I don't think this would work in resteasy reactive (I know we're using resteasy legacy) since the thread that handles the body is not guaranteed to be the same that the request one. 
I would avoid dealing with this and check the request get URI again here.